### PR TITLE
Fixes #1839 - Handle and emit dropdown key events correctly

### LINKED ISSUE #1839: ClayDropdown: On press Enter on an item doesn't fire the itemClicked event correctly

### PR DESCRIPTION
**Pull request by carloslancha**

### CODE REVIEW

**Comment by matuzalemsteles** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: packages/clay-dropdown/src/ClayDropdownBase.js [hunk 1]
```diff
@@ -214,92 +149,80 @@ class ClayDropdownBase extends ClayComponent {
...
 			originalEvent: event,
 		});
 	}
 
+	/**
+	 * Handles the item key down event
+	 * @param {!Event} event
+	 * @protected
+	 */
+	_handleItemKeyDown(event) {
```
Comment (matuzalemsteles) [inline packages/clay-dropdown/src/ClayDropdownBase.js]: hey @carloslancha should not we ignore the focus when an item is disabled?

File: packages/clay-dropdown/src/ClayDropdownBase.js [hunk 2]
```diff
@@ -14,11 +14,8 @@ import templates from './ClayDropdownBase.soy.js';
 
 const KEY_CODE_ESC = 27;
 
-const KEY_DOWN = 'ArrowDown';
```
Comment (matuzalemsteles) [inline packages/clay-dropdown/src/ClayDropdownBase.js]: Are we removing the interactions by the arrows due to UX change in MultiSelect?

Comment (carloslancha) [reply packages/clay-dropdown/src/ClayDropdownBase.js]: Yes

File: packages/clay-dropdown/src/ClayDropdownBase.js (hunk 1 above)
Comment (carloslancha) [reply packages/clay-dropdown/src/ClayDropdownBase.js]: Yes @matuzalemsteles but we can iterate over it. Could you please open an issue? Thx!

**Pull request merged**